easybuild/easyblocks/o/openssl_wrapper.py

```diff
@@ -0,0 +1,268 @@
+                os.path.join(lib_dir, 'openssl', self.openssl_engines[self.version]),
+            ]
+
+            for engines_path in lib_engines_dir:
+                if os.path.isdir(engines_path):
+                    self.ssl_sysengines = engines_path
+                    self.log.debug("Found OpenSSL engines in: %s", self.ssl_sysengines)
+
+            if not self.ssl_sysengines:
+                self.ssl_syslib = None
+                print_warning("Found OpenSSL in host system, but not its engines."
+                              "Falling back to OpenSSL in EasyBuild")
+
+        # Check system include paths for OpenSSL headers
+        cmd = "gcc -E -Wp,-v -xc /dev/null"
+        (out, ec) = run_cmd(cmd, log_all=True, simple=False, trace=False)
+
+        sys_include_dirs = []
+        for match in re.finditer(r'^\s(/[^\0\n]*)+', out, re.MULTILINE):
+            sys_include_dirs.extend(match.groups())
+        self.log.debug("Found the following include directories in host system: %s", ', '.join(sys_include_dirs))
+
+        # headers are always in "include/openssl" subdirectories
+        ssl_include_dirs = [os.path.join(include, self.name.lower()) for include in sys_include_dirs]
+        ssl_include_dirs = [include for include in ssl_include_dirs if os.path.isdir(include)]
+
+        # verify that the headers match our OpenSSL version
+        for include_dir in ssl_include_dirs:
+            opensslv_path = os.path.join(include_dir, 'opensslv.h')
+            if os.path.exists(opensslv_path):
+                opensslv = read_file(opensslv_path)
+                header_majmin_version = re.search(r"SHLIB_VERSION_NUMBER\s\"([0-9]+\.[0-9]+)", opensslv, re.M)
+                if re.match("^{}".format(*header_majmin_version.groups()), self.version):
+                    self.ssl_sysheader = include_dir
+                    self.log.info("Found OpenSSL headers in host system: %s", self.ssl_sysheader)
+                    break
+
+        if not self.ssl_sysheader:
+            self.log.info("OpenSSL headers not found in host system")
+
+    def fetch_step(self, *args, **kwargs):
+        """Fetch sources if OpenSSL component is needed"""
+        if not all([self.ssl_syslib, self.ssl_sysheader]):
+            super(EB_OpenSSL_wrapper, self).fetch_step(*args, **kwargs)
+
+    def extract_step(self):
+        """Extract sources if OpenSSL component is needed"""
+        if not all([self.ssl_syslib, self.ssl_sysheader]):
+            super(EB_OpenSSL_wrapper, self).extract_step()
+
+    def install_step(self):
+        """Symlink target OpenSSL installation"""
+        shlib_ext = get_shared_lib_ext()
+
+        if self.ssl_syslib and self.ssl_sysheader:
+            # The host system already provides the necessary OpenSSL files
+            lib_pattern = ['%s*.%s*' % (lib_so, shlib_ext) for lib_so in self.openssl_libs]
+            lib_pattern = [os.path.join(os.path.dirname(self.ssl_syslib), '%s' % ptrn) for ptrn in lib_pattern]
```

Comment:
I think this is an issue as it includes to much here, e.g.
```
OpenSSL/1.0/lib64/libcrypto.so -> /lib64/libcrypto.so
OpenSSL/1.0/lib64/libcrypto.so.10 -> /lib64/libcrypto.so.10
OpenSSL/1.0/lib64/libcrypto.so.1.0.2k -> /lib64/libcrypto.so.1.0.2k
OpenSSL/1.0/lib64/libcrypto.so.1.1 -> /lib64/libcrypto.so.1.1
OpenSSL/1.0/lib64/libcrypto.so.1.1.1g -> /lib64/libcrypto.so.1.1.1g
OpenSSL/1.0/lib64/libssl3.so -> /lib64/libssl3.so
OpenSSL/1.0/lib64/libssl.so -> /lib64/libssl.so
OpenSSL/1.0/lib64/libssl.so.10 -> /lib64/libssl.so.10
OpenSSL/1.0/lib64/libssl.so.1.0.2k -> /lib64/libssl.so.1.0.2k
OpenSSL/1.0/lib64/libssl.so.1.1 -> /lib64/libssl.so.1.1
OpenSSL/1.0/lib64/libssl.so.1.1.1g -> /lib64/libssl.so.1.1.1g
```

Also, including down to the specific version 
```
OpenSSL/1.0/lib64/libssl.so.1.0.2k -> /lib64/libssl.so.1.0.2k
```
would be dangerous since this library would disappear during an OS update.

I would say the most desirable outcome is to only have
```
libcrypto.so -> libcrypto.so.X.X  (just local symlink)
libcrypto.so.X.X -> /lib64/libcrypto.so.XX
libssl.so -> libssl.so.X.X  (just local symlink)
libssl.so.X.X -> /lib64/libssl.so.X.X
```

Maybe use something like
```
        libssl = {
            '1.0': {
                LINUX: ('libssl.so.10', 'libcrypto.so.10'),
                DARWIN: ('libssl.1.0.dylib', 'libcrypto.1.0.dylib'),
            },
            '1.1': {
                LINUX: ('libssl.so.1.1', 'libcrypto.so.1.1'),
                DARWIN: ('libssl.1.1.dylib', 'libcrypto.1.1.dylib'),
            },
        }
```
and use the absolute path (from `ctypes.cdll.LoadLibrary`) to these 2 libraries specifically instead of grabbing the directory?

Reply:
@Micket limited the extend of libraries symlinked in https://github.com/easybuilders/easybuild-easyblocks/pull/2429/commits/bcbe0372bf787e3190e4c00994eef32684dc5835